ssssort.h

```diff
@@ -81,8 +81,10 @@ static std::mt19937 gen{std::random_device{}()};
 #endif
 
 // Provides different sampling strategies to choose splitters
-template <typename Iterator, typename value_type>
+template <typename Iterator>
 struct Sampler {
+    using value_type = typename std::iterator_traits<Iterator>::value_type;
```

Comment:
Is that guaranteed to work? I wasn't sure, that's why I dragged `value_type` along